### PR TITLE
Replace // ... with /* ... */

### PR DESCRIPTION
**Pull request by gus-massa**
Replace // ... with /* ... */ to support old C compilers.

I think it should be merged in 6.5.

### CODE REVIEW

File: racket/src/racket/src/optimize.c [hunk 1]
```diff
@@ -2563,9 +2563,10 @@ static Scheme_Object *rator_implies_predicate(Scheme_Object *rator, int argc)
       return scheme_list_pair_p_proc;
     else if (SAME_OBJ(rator, scheme_mcons_proc))
       return scheme_mpair_p_proc;
-    // XXX This could be implemented
-    // else if (SAME_OBJ(rator, scheme_make_byte_string_p))
-    //  return scheme_byte_string_p_proc;
+    /* This could be implemented
```
Comment (stamourv) [inline racket/src/racket/src/optimize.c]: Nitpick: Your changes dropped the `XXX`. Is that a keyword that someone may grep for?

Comment (rfindler) [inline racket/src/racket/src/optimize.c]: I think Jay uses that?​

**Comment by gus-massa** [comment]
I restored the `XXX`.

**Comment by mflatt** [comment]
(I don't know that "XXX" has any meaning here, but I'm interested to hear if someone specifically wants to keep it.)

I thought that `//` comments would create trouble for the MSVC build, but they didn't. Maybe it was a problem back with MSVC 2005 (instead of 2008). So, even though I still prefer using only `/* ... */`, maybe enough time has passed that we should start allowing `//`.

**Comment by gus-massa** [comment]
OK, let's close this.

**Pull request closed without merging**